index.android.js

```diff
@@ -1,19 +1,34 @@
-
-import { NativeModules } from 'react-native'
+import { NativeEventEmitter, NativeModules } from 'react-native'
 
 const { RNPassportReader } = NativeModules
+const Emitter = new NativeEventEmitter(RNPassportReader)
+
 const DATE_REGEX = /^\d{6}$/
 
+const constants = RNPassportReader.getConstants()
+
 module.exports = {
   ...RNPassportReader,
+  constants,
   scan
 }
 
-function scan ({ documentNumber, dateOfBirth, dateOfExpiry, quality=1 }) {
+async function scan ({ documentNumber, dateOfBirth, dateOfExpiry, quality = 1, onProgress = () => {} }) {
   assert(typeof documentNumber === 'string', 'expected string "documentNumber"')
   assert(isDate(dateOfBirth), 'expected string "dateOfBirth" in format "yyMMdd"')
   assert(isDate(dateOfExpiry), 'expected string "dateOfExpiry" in format "yyMMdd"')
-  return RNPassportReader.scan({ documentNumber, dateOfBirth, dateOfExpiry, quality })
+  
+  const listener = Emitter.addListener(constants.EVENT_NFC_PROGRESS, onProgress)
+  
+  try {
+    const result = await RNPassportReader.scan({ documentNumber, dateOfBirth, dateOfExpiry, quality })
+    
+    listener.remove()
+    return result
+  } catch (error) {
+    listener.remove()
+    throw error
+  }
```

Comment:
slightly simpler pattern:
```suggestion
  try {
    return await RNPassportReader.scan({ documentNumber, dateOfBirth, dateOfExpiry, quality })
  } finally {
    listener.remove()
  }
```